### ISSUE TITLE
TextNode

### ISSUE THREAD

**Issue by mP1** [mP1/walkingkooka#1425]
- Text holds a string value without style
- includes Style contains that can hold more Text and style properties key/value pairs.
- TextNodeReferences can hold references to other external things.
- TextNodeCollection holds 1 or more TextNodes.
- implements Node
- implements HasSearchNode.

**Comment by mP1** [comment]
```
abstract **TextNode** implements Node {
   class **Text** {
      String value;
   },
    class **TextProperties** {
       Map<TextAttributeName, Object> attributes; // styles etc.
       List<TextNode> children
    },
    class **TextStyleName** extends Reference {
    }
}
```

**Comment by mP1** [comment]
Closed by #1477
